### PR TITLE
chore: update flag and policy descriptions

### PR DESCRIPTION
**Pull request by markmichon**
## Description

- Light update to flags to match format.
- Makes the policy descriptions more detailed and includes recommended action for each.

todo:
- [ ] update snapshots

## Checklist

- [ ] I've added test coverage that shows my fix or feature works as expected.
- [ ] I've updated or added documentation if required.
- [ ] I've included usage information in the description if CLI behavior was updated or added.
- [x] PR title follows [Conventional Commits](https://www.conventionalcommits.org/) format

### CODE REVIEW

**Review by elsapet — APPROVED** [review]
Going forward should we ping you @markmichon on PRs that add new policies to the policies.yml file?

File: pkg/commands/process/settings/policies.yml [hunk 1]
```diff
@@ -43,7 +43,7 @@ ssl_certificate_verification_disabled:
     - path: policies/ssl_certificate_verification_disabled.rego
       name: bearer.ssl_certificate_verification_disabled
 application_level_encryption_missing:
-  description: "Application level encryption missing"
+  description: "Application level encryption missing. Enable application level encryption to reduce the risk of leaking sensitive data."
```
Comment (spdawson) [inline pkg/commands/process/settings/policies.yml:57]: Should "application level" by hyphenated, as "application-level"? After all, we are talking about encryption at the level of the application, and not encryption of one or more "application levels", which is what the unhyphenated form strictly suggests.

File: pkg/commands/process/settings/policies.yml [hunk 2]
```diff
@@ -54,7 +54,7 @@ application_level_encryption_missing:
     - path: policies/application_level_encryption.rego
       name: bearer.application_level_encryption
 insecure_smtp_processing_sensitive_data:
-  description: "Communication with insecure SMTP in an application processing sensitive data"
+  description: "Communication with insecure SMTP in an application processing sensitive data. Verify that SMTP settings use OpenSSL or equivalent."
```
Comment (spdawson) [inline pkg/commands/process/settings/policies.yml]: Maybe "using insecure SMTP" is preferable to "with insecure SMTP". "Communication with insecure SMTP" sounds as though "insecure SMTP" were the recipient of communication, rather than the means of communication.

File: pkg/commands/process/settings/policies.yml [hunk 3]
```diff
@@ -98,7 +98,7 @@ insecure_ftp_processing_sensitive_data:
     - path: policies/insecure_ftp.rego
       name: bearer.insecure_ftp
 insecure_http_get:
-  description: "Communicating with insecure HTTP GET in an application processing sensitive data"
+  description: "Communicating with insecure HTTP GET in an application processing sensitive data. Ensure all HTTP communication occurs over HTTPS."
```
Comment (spdawson) [inline pkg/commands/process/settings/policies.yml]: Maybe "using insecure HTTP GET" is preferable to "with insecure HTTP GET" here?

File: pkg/commands/process/settings/policies.yml [hunk 4]
```diff
@@ -109,7 +109,7 @@ insecure_http_get:
     - path: policies/insecure_http_get.rego
       name: bearer.insecure_http_get
 insecure_http_with_data_category:
-  description: "Communicating Data Category with insecure HTTP"
+  description: "Communicating Data Category with insecure HTTP. Ensure all HTTP communication occurs over HTTPS."
```
Comment (spdawson) [inline pkg/commands/process/settings/policies.yml]: Maybe "using insecure HTTP" rather than "with insecure HTTP" here.

File: pkg/flag/general_flags.go [hunk 5]
```diff
@@ -5,7 +5,7 @@ var (
 		Name:            "config-file",
 		ConfigName:      "config-file",
 		Value:           "",
-		Usage:           "Load configuration from the defined path",
+		Usage:           "Load configuration from the defined path.",
```
Comment (spdawson) [inline pkg/flag/general_flags.go]: Maybe "specified" is better than "defined"?

File: pkg/flag/policy_flags.go [hunk 6]
```diff
@@ -5,13 +5,13 @@ var (
 		Name:       "skip-policy",
 		ConfigName: "policy.skip-policy",
 		Value:      []string{},
-		Usage:      "specify the comma separated ids of the policies you would like to skip. Runs all other policies.",
+		Usage:      "Specify the comma separated ids of the policies you would like to skip. Runs all other policies.",
```
Comment (spdawson) [inline pkg/flag/policy_flags.go]: Should "comma separated" be hyphenated?

File: pkg/flag/policy_flags.go [hunk 7]
```diff
@@ -5,13 +5,13 @@ var (
...
 	}
 	OnlyPolicyFlag = Flag{
 		Name:       "only-policy",
 		ConfigName: "policy.only-policy",
 		Value:      []string{},
-		Usage:      "specify the comma separated ids of the policies you would like to run. Skips all other policies.",
+		Usage:      "Specify the comma separated ids of the policies you would like to run. Skips all other policies.",
```
Comment (spdawson) [inline pkg/flag/policy_flags.go]: Should "comma separated" be hyphenated?

File: pkg/flag/process_flags.go [hunk 8]
```diff
@@ -6,7 +6,7 @@ var (
 		ConfigName: "process.port",
 		Shorthand:  "p",
 		Value:      "",
-		Usage:      "server listening on what port",
+		Usage:      "Set the process server listening port.",
```
Comment (spdawson) [inline pkg/flag/process_flags.go]: Maybe "Set the listening port of the [process] server"? Does the word "process" add value here, or confusion? (Is the server serving processes?)

File: pkg/flag/report_flags.go [hunk 9]
```diff
@@ -24,13 +24,13 @@ var (
 		Name:       "report",
 		ConfigName: "report.report",
 		Value:      ReportDetectors,
-		Usage:      "Specify the kind of report (detectors, dataflow, policies, stats)",
+		Usage:      "Specify the kind of report (detectors, dataflow, policies, stats).",
```
Comment (spdawson) [inline pkg/flag/report_flags.go]: Maybe "type" is preferable to "kind" here?

File: pkg/flag/repository_flags.go [hunk 10]
```diff
@@ -5,19 +5,19 @@ var (
 		Name:       "branch",
 		ConfigName: "repository.branch",
 		Value:      "",
-		Usage:      "pass the branch name to be scanned",
+		Usage:      "Pass the branch name to be scanned.",
```
Comment (spdawson) [inline pkg/flag/repository_flags.go]: Maybe "specify" is preferable to "pass", for consistency?

File: pkg/flag/worker_flags.go [hunk 11]
```diff
@@ -9,25 +9,25 @@ var (
...
 		Value:      1,
-		Usage:      "number of processing workers to spawn",
+		Usage:      "The number of processing workers to spawn.",
 	}
 	TimeoutFlag = Flag{
 		Name:       "timeout",
 		ConfigName: "worker.timeout",
 		Value:      10 * time.Minute,
-		Usage:      "time allowed to complete scan",
+		Usage:      "The max time alloted to complete the scan.",
```
Comment (spdawson) [inline pkg/flag/worker_flags.go]: "Maximum" is probably better than "max" here.

File: pkg/flag/worker_flags.go [hunk 12]
```diff
@@ -9,25 +9,25 @@ var (
...
 	}
 	TimeoutFileMinimumFlag = Flag{
 		Name:       "timeout-file-min",
 		ConfigName: "worker.timeout-file-min",
 		Value:      5 * time.Second,
-		Usage:      "minimum timeout assigned to scanning file, this config superseeds timeout-second-per-bytes",
+		Usage:      "Minimum timeout assigned for scanning each file. This config superseeds timeout-second-per-bytes.",
```
Comment (spdawson) [inline pkg/flag/worker_flags.go:24]: "Configuration" is probably better than "config" here, and elsewhere &mdash; unless brevity is essential due to a space/formatting constraint.

File: pkg/flag/worker_flags.go [hunk 13]
```diff
@@ -39,31 +39,31 @@ var (
...
 		Value:      60 * time.Second,
-		Usage:      "maximum time for worker process to come online",
+		Usage:      "Maximum time to wait for a worker process to come online.",
 	}
 	FileSizeMaximumFlag = Flag{
 		Name:       "file-size-max",
 		ConfigName: "worker.file-size-max",
 		Value:      100 * 1000, // 100 KB
-		Usage:      "ignore files with file size larger than this config",
+		Usage:      "Ignore files with a file size larger than this setting.",
```
Comment (spdawson) [inline pkg/flag/worker_flags.go]: I think we can shorten this to "ignore files larger than the specified value."

File: pkg/flag/worker_flags.go [hunk 14]
```diff
@@ -39,31 +39,31 @@ var (
...
 	}
 	FilesToBatchFlag = Flag{
 		Name:       "files-to-batch",
 		ConfigName: "worker.files-to-batch",
 		Value:      1,
-		Usage:      "number of files to batch to worker",
+		Usage:      "Define the number of files to batch per worker.",
```
Comment (spdawson) [inline pkg/flag/worker_flags.go]: Maybe prefer "specify" to "define", for consistency.

File: pkg/flag/worker_flags.go [hunk 15]
```diff
@@ -39,31 +39,31 @@ var (
...
 	}
 	MemoryMaximumFlag = Flag{
 		Name:       "memory-max",
 		ConfigName: "worker.memory-max",
 		Value:      800 * 1000 * 1000, // 800 MB
-		Usage:      "if memory needed to scan a file surpasses this limit, skip the file",
+		Usage:      "If the memory needed to scan a file surpasses this limit, skip the file.",
```
Comment (spdawson) [inline pkg/flag/worker_flags.go]: Maybe change "this limit" to "the specified limit"?

File: pkg/flag/worker_flags.go [hunk 16]
```diff
@@ -39,31 +39,31 @@ var (
...
 	}
 	ExistingWorkerFlag = Flag{
 		Name:       "existing-worker",
 		ConfigName: "worker.existing-worker",
 		Value:      "",
-		Usage:      "URL of an existing worker",
+		Usage:      "Set the URL of an existing worker.",
```
Comment (spdawson) [inline pkg/flag/worker_flags.go]: Prefer "specify" to "set", for consistency.

**Review by spdawson — APPROVED** [review]
Looks very good in general. I've added a few suggestions, but these are mainly small grammar things, and places where the language could be more consistent. Feel free to ignore :slightly_smiling_face:

**Comment by markmichon** [comment]
> Going forward should we ping you @markmichon on PRs that add new policies to the policies.yml file?

@elsapet I'll try to keep an eye on them, but that would be great. They don't need to be blocked by me though (particularly pre-launch).

**Comment by markmichon** [comment]
@vjerci or @cfabianski, does this basically mean something like: "If the bytes per second is lower than this value, timeout and skip the file"? Essentially setting a minimum speed requirement? or something else? I'm not sure I understand it from the description.
```
Name:       "timeout-file-second-per-bytes",
		ConfigName: "worker.timeout-file-second-per-bytes",
		Value:      10 * 1000, // 10kb/s
		Usage:      "number of file size bytes producing a second of timeout assigned to scanning a file",
```

**Comment by cfabianski** [comment]
> @vjerci or @cfabianski, does this basically mean something like: "If the bytes per second is lower than this value, timeout and skip the file"? Essentially setting a minimum speed requirement? or something else? I'm not sure I understand it from the description.

I'll let @vjerci double check that :)
